### PR TITLE
Feature/mds 3196 intake and manage permit required reports

### PR DESCRIPTION
**Pull request by mhuseinov**
Users have the ability to:

-  create/edit a permit required reports
-  filter reports
-  file history support
-  comments 
-  added labels for the reports pages

Fix for Code Required Reports filtering

image
image
image

### CODE REVIEW

File: services/minespace-web/common/actionCreators/reportActionCreator.js [hunk 1]
```diff
@@ -48,11 +48,17 @@ export const createMineReport = (mineGuid, payload) => (dispatch) => {
...
+  mineGuid,
+  reportsType = Strings.MINE_REPORTS_TYPE.codeRequiredReports
+) => (dispatch) => {
   dispatch(request(reducerTypes.GET_MINE_REPORTS));
   dispatch(showLoading());
   return CustomAxios()
-    .get(`${ENVIRONMENT.apiUrl}${API.MINE_REPORTS(mineGuid)}`, createRequestHeader())
+    .get(
+      `${ENVIRONMENT.apiUrl}${API.MINE_REPORTS(mineGuid)}?mine_reports_type=${reportsType}`,
+      createRequestHeader()
```
Comment (mmumby) [inline services/minespace-web/common/actionCreators/reportActionCreator.js:60]: Can you move this into the MINE_REPORTS const as another param?
API.MINE_REPORTS(mineGuid, reportType)

File: services/core-web/src/constants/routes.js [hunk 2]
```diff
@@ -160,6 +161,13 @@ export const MINE_REPORTS = {
   component: MineReportInfo,
 };
 
+export const MINE_PERMIT_REQUIRED_REPORTS = {
+  route: "/mine-dashboard/:id/reports/mine-required-reports",
+  dynamicRoute: (id, filterParams) =>
```
Comment (mmumby) [inline services/core-web/src/constants/routes.js:166]: can we name this permit-require-reports? for consistency of tab and URL name

Comment (mmumby) [reply services/core-web/src/constants/routes.js:166]: disregard

File: services/core-api/app/api/mines/permits/permit_conditions/models/permit_condition_category.py [hunk 3]
```diff
@@ -24,4 +24,8 @@ def __repr__(self):
 
     @classmethod
     def get_all(cls):
-        return cls.query.order_by(cls.display_order).all()
\ No newline at end of file
+        return cls.query.order_by(cls.display_order).all()
+
+    @classmethod
+    def find_by_permit_condition_category_code(cls, code):
+        return cls.query.filter_by(condition_category_code=code).one_or_none()
```
Comment (JustinMacaulay) [inline services/core-api/app/api/mines/permits/permit_conditions/models/permit_condition_category.py:31]: both of these should probably have an active_ind check on them.

File: services/core-api/app/api/mines/reports/models/mine_report.py [hunk 4]
```diff
@@ -126,10 +141,40 @@ def find_by_mine_report_guid(cls, _id):
     def find_by_mine_guid_and_category(cls, _id, category):
         try:
             uuid.UUID(_id, version=4)
-            reports = cls.query.filter_by(mine_guid=_id).all()
+            reports = cls.query.filter(
+                MineReport.permit_condition_category_code.is_(None)).filter_by(mine_guid=_id).all()
```
Comment (JustinMacaulay) [inline services/core-api/app/api/mines/reports/models/mine_report.py:145]: should have a check for deleted_ind

File: services/core-api/app/api/mines/reports/resources/mine_reports.py [hunk 5]
```diff
@@ -62,25 +64,47 @@ def post(self, mine_guid):
...
 
         data = self.parser.parse_args()
-
-        mine_report_definition = MineReportDefinition.find_by_mine_report_definition_guid(
-            data['mine_report_definition_guid'])
-        permit = Permit.find_by_permit_guid_or_no(data['permit_guid'])
-        if mine_report_definition is None:
-            raise BadRequest('A report must be selected from the list.')
+        permit_condition_type_code = data.get('permit_condition_category_code', None)
+        current_app.logger.debug(permit_condition_type_code)
```
Comment (JustinMacaulay) [inline services/core-api/app/api/mines/reports/resources/mine_reports.py:68]: if not needed please remove.

File: services/core-api/tests/factories.py [hunk 6]
```diff
@@ -324,6 +330,17 @@ def mine_report_submissions(obj, create, extracted, **kwargs):
 
         MineReportSubmissionFactory.create_batch(size=extracted, report=obj, **kwargs)
 
+    # @factory.post_generation
```
Comment (JustinMacaulay) [inline services/core-api/tests/factories.py:333]: Remove if not needed please.

**Review by JustinMacaulay — COMMENTED** [review]
Just a few comments.

File: services/core-web/src/components/Forms/reports/AddMinePermitRequiredForm.js [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+                id="permit_condition_category_code"
+                name="permit_condition_category_code"
+                label="Report Type*"
+                placeholder="Select report type"
+                data={this.props.dropdownPermitConditionCategoryOptions}
+                doNotPinDropdown
+                component={renderConfig.SELECT}
+                validate={[required]}
+                format={null}
+              />
```
Comment (mmumby) [inline services/core-web/src/components/Forms/reports/AddMinePermitRequiredForm.js:120]: remove format={null}

File: services/core-web/src/components/Forms/reports/AddMinePermitRequiredForm.js [hunk 8]
```diff
@@ -0,0 +1,215 @@
+/* eslint-disable */
```
Comment (mmumby) [inline services/core-web/src/components/Forms/reports/AddMinePermitRequiredForm.js:1]: remove

File: services/minespace-web/common/actionCreators/reportActionCreator.js (hunk 1 above)
Comment (mhuseinov) [reply services/minespace-web/common/actionCreators/reportActionCreator.js:60]: done

File: services/core-web/src/components/Forms/reports/AddMinePermitRequiredForm.js (hunk 7 above)
Comment (mhuseinov) [reply services/core-web/src/components/Forms/reports/AddMinePermitRequiredForm.js:120]: done

File: services/core-api/tests/factories.py (hunk 6 above)
Comment (mhuseinov) [reply services/core-api/tests/factories.py:333]: removed

File: services/core-api/app/api/mines/reports/models/mine_report.py (hunk 4 above)
Comment (mhuseinov) [reply services/core-api/app/api/mines/reports/models/mine_report.py:145]: delete works as a hard delete for reports

File: services/core-api/app/api/mines/permits/permit_conditions/models/permit_condition_category.py (hunk 3 above)
Comment (mhuseinov) [reply services/core-api/app/api/mines/permits/permit_conditions/models/permit_condition_category.py:31]: maybe not for the get_all, but I'll definitely add one for find_by_permit_condition_category_code

File: services/core-api/app/api/mines/reports/resources/mine_reports.py (hunk 5 above)
Comment (mhuseinov) [reply services/core-api/app/api/mines/reports/resources/mine_reports.py:68]: done

File: services/core-web/src/components/Forms/reports/AddMinePermitRequiredForm.js (hunk 8 above)
Comment (mhuseinov) [reply services/core-web/src/components/Forms/reports/AddMinePermitRequiredForm.js]: done

**Review by JustinMacaulay — APPROVED** [review]
Looks good to me!

**Pull request merged**